core/src/main/java/org/apache/accumulo/testing/core/continuous/ContinuousIngest.java

```diff
@@ -45,11 +45,43 @@
   private static final byte[] EMPTY_BYTES = new byte[0];
 
   private static List<ColumnVisibility> visibilities;
+  private static long lastPause;
+  private static long pauseWaitMs;
 
   private static ColumnVisibility getVisibility(Random rand) {
     return visibilities.get(rand.nextInt(visibilities.size()));
   }
 
+  private static boolean pauseEnabled(Properties props) {
+    String value = props.getProperty(TestProps.CI_INGEST_PAUSE_ENABLED);
+    return value != null && value.equals("true");
+  }
+
+  private static int getPauseWaitMs(Properties props, Random rand) {
+    int waitMin = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_WAIT_MIN));
+    int waitMax = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_WAIT_MAX));
+    return (rand.nextInt(waitMax - waitMin) + waitMin) * 1000;
+  }
+
+  private static int getPauseDurationMs(Properties props, Random rand) {
+    int durationMin = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_DURATION_MIN));
+    int durationMax = Integer.parseInt(props.getProperty(TestProps.CI_INGEST_PAUSE_DURATION_MAX));
+    return (rand.nextInt(durationMax - durationMin) + durationMin) * 1000;
+  }
+
+  private static void pauseCheck(Properties props, Random rand) throws InterruptedException {
+    if (pauseEnabled(props)) {
+      if (System.currentTimeMillis() > (lastPause + pauseWaitMs)) {
+        long pauseDurationMs = getPauseDurationMs(props, rand);
+        System.out.println("Pausing ingest for " + pauseDurationMs + " ms - " + System.currentTimeMillis());
+        Thread.sleep(pauseDurationMs);
+        lastPause = System.currentTimeMillis();
+        pauseWaitMs = getPauseWaitMs(props, rand);
+        System.out.println("Ingest restarted. Will pause again in " + pauseWaitMs + " ms at " + (lastPause + pauseWaitMs));
```

Comment:
Is STDOUT consistent with the rest of CI output, or should we use a logger?

Reply:
STDOUT was consistent but I think a logger is better so I switched everything.